### DISCUSSION TITLE
`machine-id`, `boot_id`

### DISCUSSION THREAD

**Discussion by beroal** [roddhjav/apparmor.d#500] [Q&A]
Should I allow or deny access to the following files?

```
/etc/machine-id
@{PROC}/sys/kernel/random/boot_id
```

I see that some profiles allow it and some deny. The program I'm working on is pavucontrol-qt.

**Comment by roddhjav** [comment]
This mostly depends of your security objectives.
* `/etc/machine-id` is a uniquely identify your machine. It is is generated on first boot and does not change. It is required for most dbus call (and part of the `bus-session` abs. Denying it is safe for some app, but will crash other... It can leak information about your activity this is why it is usually set to `b08dfa6083e7567a1921a715000001fb` on all anonymity distribution (whonix, tail).
* `@{PROC}/sys/kernel/random/boot_id` uniquely identify your boot. It is included in the `nameservice-strict` abs. I would say, it is "safe" and mostly used by systemd for log/debug.

To sum up, keep them allowed if they are needed.
